File: Week 6/sentimental-readability/readability.py
# TODO
words = input("Text: ")
swords = words.split()
sen = words.count(".")
sen += words.count("!")
sen += words.count("?")

s = (sen/len(swords)) * 100
l = 0
for i in swords:
    if "." in i or "!" in i or "?" in i:
        l-=1
    l+= len(i)
l /= len(swords)
l*=100
grade = 0.0588 * l - 0.296 * s - 15.7
if grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
else:
    print("Grade",int(grade))

File: Week 2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");
    double letters = 0;
    double sen = 0;
    double word = 1;
    for (int i = 0; i < strlen(text); i++)
    {

        if (text[i] == '!' || text[i] == '.' || text[i] == '?')
        {
            sen++;
        }
        else if (text[i] != ' ')
        {
            letters++;
        }
        else
        {
            word++;
        }
    }

    sen = (sen / word ) * 100;
    letters = (letters / word) * 100;
    double index = 0.0588 * letters - 0.296 * sen - 15.7;
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 15)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int)index);
    }
}
